controllers/operands/wasp.go

```diff
@@ -0,0 +1,449 @@
+package operands
+
+import (
+	"errors"
+	"maps"
+	"os"
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+
+	sdkapi "kubevirt.io/controller-lifecycle-operator-sdk/api"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	hcov1beta1 "github.com/kubevirt/hyperconverged-cluster-operator/api/v1beta1"
+	"github.com/kubevirt/hyperconverged-cluster-operator/controllers/common"
+	hcoutil "github.com/kubevirt/hyperconverged-cluster-operator/pkg/util"
+)
+
+const (
+	// For simplicity matters, re-use the virt-handler SA. That will spare
+	// the HCO controller from deploying custom privileged SA with SCC.
+	waspServiceAccount = "kubevirt-handler"
+	// Set dry run mode for e2e testing purposes
+	waspDryRunAnnotation = "wasp.hyperconverged.io/dry-run"
+)
+
+func newWaspHandler(Client client.Client, Scheme *runtime.Scheme) Operand {
+	return &conditionalHandler{
+		operand: &genericOperand{
+			Client: Client,
+			Scheme: Scheme,
+			crType: "DaemonSet",
+			hooks:  &waspHooks{},
+		},
+		getCRWithName: func(hc *hcov1beta1.HyperConverged) client.Object {
+			return NewWaspWithNameOnly(hc)
+		},
+		shouldDeploy: func(hc *hcov1beta1.HyperConverged) bool {
+			// convert from anon to private
+			return hc.Spec.FeatureGates.EnableHigherDensityWithSwap != nil &&
+				*hc.Spec.FeatureGates.EnableHigherDensityWithSwap
```

Comment:
Let's make a function from it, and use it also for the ClusterRoleHandler and the and the ClusterRoleBindingHandler, so we'll have this common logic in one place.